### PR TITLE
[DEV-1418] Allow fetching from the CMS the "In evidenza" section on the home page

### PR DESCRIPTION
**Pull request by marcobottaro**
Depends on #666

#### List of Changes

Allow fetching from the CMS the "In evidenza" section on the home page and fix distances

#### How Has This Been Tested?

It has been tested by setting `FETCH_FROM_STRAPI=true` environment variable and by replacing
`        image:
            item.attributes.image.data && item.attributes.image.data.attributes,`
with

```
          image: {
            name:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.name) ||
              '',
            alternativeText:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.alternativeText) ||
              '',
            width:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.width) ||
              0,
            height:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.height) ||
              0,
            ext:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.ext) ||
              '',
            mime:
              (item.attributes.image.data &&
                item.attributes.image.data.attributes.mime) ||
              '',
            url:
              (item.attributes.image.data &&
                `http://localhost:1337${item.attributes.image.data.attributes.url}`) ||
              '',
          },
```

in `lib/homepage.ts` to see the images coming from Strapi's local instance

#### Screenshots (if appropriate):
image

From CMS:
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tommaso1 — APPROVED** [review]

File: apps/nextjs-website/src/components/organisms/News/NewsShowcase.tsx [hunk 1]
```diff
@@ -5,48 +5,57 @@ import { Box } from '@mui/material';
...
-      title: string;
+    link: {
+      url: string;
+      text: string;
     };
   }[];
 };
 
-const News = ({ title, subtitle, cta, marginTop, cards }: NewsProps) => {
+const NewsShowcase = ({
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/organisms/News/NewsShowcase.tsx:31]: Also Folder should be renamed accordingly `apps/nextjs-website/src/components/organisms/NewsShowcase`

File: apps/nextjs-website/src/lib/homepage.ts [hunk 2]
```diff
@@ -83,9 +104,9 @@ export const makeHomepagePropsFromStatic = (
     boldTitle: staticHeader.boldTitle,
     cards: staticHomepage.heroItems,
   },
-  news: {
-    title: staticHomepage.news.title,
-    cards: staticHomepage.news.list,
+  newsShowcase: {
+    title: staticHomepage.newsShowcase.title,
+    items: staticHomepage.newsShowcase.items,
   },
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/homepage.ts:110]:
```suggestion
  newsShowcase: staticHomepage.newsShowcase,
```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

**Comment by marcobottaro** [comment]
> I've add the `depends on` with PR #666.
> 
> Similar to [another PR where I made this comment](https://github.com/pagopa/developer-portal/pull/632#issuecomment-1941967156), I think you should add the dependency to the PR that adds the structure on Strapi.
> 
> So, IMHO, or you always add the `depends on` on the PR, or you create a single PR that both add the structure on Strapi and update the NextJS page fetching content from Strapi (in this case you don't need to specify the `depends on`). My assumption in this last option is that the PR is small enough.

Even if - in theory - this FE-only PR does not need the PR #666 to be merged, I agree with adding the dependency!

**Review by MarcoPonchia — APPROVED** [review]

File: apps/nextjs-website/src/lib/homepage.ts [hunk 3]
```diff
@@ -8,21 +8,26 @@ export type HomepageProps = {
...
-  readonly news: {
+  readonly newsShowcase: {
     readonly title: string;
-    readonly cards: readonly {
+    readonly items: readonly {
       readonly comingSoon?: boolean;
       readonly title: string;
-      readonly dateString?: string;
-      readonly image?: {
+      readonly publishedAt?: string;
```
Comment (datalek) [inline apps/nextjs-website/src/lib/homepage.ts:16]: Can be a `Date` instead of string?

File: apps/nextjs-website/src/lib/strapi/homepage.ts [hunk 4]
```diff
@@ -34,13 +36,32 @@ const ProductCodec = t.strict({
   }),
 });
 
+const NewsItemCodec = t.strict({
+  attributes: t.strict({
+    comingSoon: t.boolean,
+    title: t.string,
+    link: LinkCodec,
+    publishedAt: t.string,
```
Comment (datalek) [inline apps/nextjs-website/src/lib/strapi/homepage.ts:44]: For this suggestion you have to import `io-ts-types` as following `import * as tt from 'io-ts-types';`

```suggestion
    publishedAt: tt.DateFromISOString,
```

**Review by datalek — APPROVED** [review]

**Comment by kin0992** [comment]
> > I've add the `depends on` with PR #666.
> > Similar to [another PR where I made this comment](https://github.com/pagopa/developer-portal/pull/632#issuecomment-1941967156), I think you should add the dependency to the PR that adds the structure on Strapi.
> > So, IMHO, or you always add the `depends on` on the PR, or you create a single PR that both add the structure on Strapi and update the NextJS page fetching content from Strapi (in this case you don't need to specify the `depends on`). My assumption in this last option is that the PR is small enough.
> 
> Even if - in theory - this FE-only PR does not need the PR #666 to be merged, I agree with adding the dependency!

Yep, but how can it work fetching the content from Strapi if the codebase doesn't have the concept of that new structure (because the code is within another PR)? That's why IMHO it's easier to have one single PR with the whole work 😄

File: apps/nextjs-website/src/lib/homepage.ts (hunk 3 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/homepage.ts]: Yes!

File: apps/nextjs-website/src/lib/strapi/homepage.ts (hunk 4 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/strapi/homepage.ts]: Nice

**Review by datalek — APPROVED** [review]

**Pull request merged**